orbit-android/src/main/java/com/babylon/orbit/OrbitViewModel.kt

```diff
@@ -16,15 +16,20 @@
 
 package com.babylon.orbit
 
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
 import androidx.lifecycle.LifecycleOwner
 import androidx.lifecycle.ViewModel
-import com.uber.autodispose.android.lifecycle.autoDispose
 import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.plusAssign
 
 open class OrbitViewModel<STATE : Any, SIDE_EFFECT : Any>(
     private val container: OrbitContainer<STATE, SIDE_EFFECT>
 ) : ViewModel() {
 
+    private val disposables = CompositeDisposable()
```

Comment:
This does not scale actually. I've recently done a prototype composite screen with multiple widgets where one of them was hosted in a shared view model and multiple observers.

Doing it this way will unsubscribe all observers at the same time with the one unsubscribing the earliest. Each observer should be unsubscribed independently of the others instead. 

I suggest creating a lifecycle observer class to store the disposable it's responsible for disposing of.

Reply:
Good point, I did not consider multiple views per view model case.